### PR TITLE
build fixes for openblas

### PR DESCRIPTION
**Pull request by lee218llnl**
Openblas was failing to build on our CHAOS 5 x86_64 clusters. One issue is that it was trying to add AVX2 instructions for a platform that did not have it, so I made avx2 a variant that is off by default. I am open to this being on by default if others think we should. I was also getting link errors during the build. This appears to be due to the parallel build, so I added "parallel = false".

### CODE REVIEW

**Comment by tgamblin** [comment]
@lee218llnl: is `avx2` auto detected by the OpenBLAS build or does it have to be set manually?  It would be nice if this were just tied to the architecture and didn't have to be a manual variant

**Comment by davydden** [comment]
>  One issue is that it was trying to add AVX2 instructions for a platform that did not have it

would you mind reporting it [upstream](https://github.com/xianyi/OpenBLAS/issues). 

I would expect openblas to be clever enough to do such things automatically, but if it is not robust enough, then we would need to have some workaround. Is there anything one could use from Python to avoid a variant and control vectorization internally in `package.py`?

File: var/spack/repos/builtin/packages/openblas/package.py [hunk 1]
```diff
@@ -84,6 +88,9 @@ def install(self, spec, prefix):
 
         make_targets = ['libs', 'netlib']
 
+        if '+avx2' not in spec:
```
Comment (davydden) [inline var/spack/repos/builtin/packages/openblas/package.py]: should not that be `~avx2`?

**Comment by becker33** [comment]
Couldn't we do this using the `target` attribute in the architecture? We would have to improve the autodetection of targets to something more specific than x86_64, but then we could use it to determine appropriate compiler flags

**Comment by adamjstewart** [comment]
So, let me throw a wrench into things real quick. We have a cluster with both Sandy Bridge (AVX) and Haswell (AVX2) nodes. I generally build things on the AVX nodes, but want it to also run on the AVX2 nodes. I do this by specifying the appropriate compiler flags in `compilers.yaml`, as mentioned in #974 so that it builds instruction sets for both architectures. So, I would like this to continue to work, especially now that we're about to get some new KNL nodes with a third architecture in the mix.

**Comment by lee218llnl** [comment]
It turns out that the AVX2 issue is that older gcc's do not support the vpermpd instruction. I modified the package to disable AXX2 for gcc compilers equal to or prior to 4.8.4. I haven't exhaustively tested compiler versions, but I found that 4.8.5 works OK and 4.8.2 does not work here at LLNL.

**Comment by tgamblin** [comment]
@adamjstewart: I think your changes will continue to work with @lee218llnl's updates, at least as well as they did before. 

@lee218llnl: does @adamjstewart need to worry about the flags with this change, or will building a multi-arch binary like he does work ok?  It sounds like this is an issue with `gcc`'s `vpermpd` intrinsic and adding `-march=haswell` wouldn't really affect it.

**Comment by lee218llnl** [comment]
@tgamblin and @adamjstewart, I'm not sure how this will work with @adamjstewart's multi-arch binary. The file that was causing me grief contained assembly code, so multi-arch compiler flags wouldn't help if the resulting object file is linked in. I haven't looked into the openblas source enough to know how much they do directly in assembly versus relying on compiler optimizations. My guidance would be to build on the oldest architecture you want to support, but perhaps if you're willing to dive into openblas a bit more, then you may find a more ideal solution.

**Comment by lee218llnl** [comment]
Don't merge yet, I tried resolving a conflict and it broke... working on a fix.

**Comment by lee218llnl** [comment]
OK, I think it's good now.

**Pull request merged**